chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,2319 @@
-                "9 digit NMEA", "6 digit Magellan", "DGPS"],
-    }
-
-    def get_features(self):
-        rf = chirp_common.RadioFeatures()
-        rf.has_settings = True
-        rf.has_dtcs = False
-        rf.has_dtcs_polarity = False
-        rf.has_bank = False
-        rf.has_mode = True
-        rf.has_tuning_step = False
-        rf.can_odd_split = True
-        rf.valid_duplexes = ["", "-", "+", "split"]
-        rf.valid_modes = list(MODES.values())
-        rf.valid_tmodes = ["", "Tone", "TSQL"]
-        rf.valid_characters = \
-            chirp_common.CHARSET_ALPHANUMERIC + "/.-+*)('&%$#! ~}|{"
-        rf.valid_name_length = 7
-        rf.valid_tuning_steps = STEPS
-        rf.memory_bounds = (0, self._upper)
-        return rf
+    _ARG_DELIMITER = " "
+    _BAUDS = [9600]
+    _CMD_DELIMITER = "\r"
+    _DISABLE_AI_CMD = ('AI', '0')
+    _DUPLEX = {0: "", 1: "+", 2: "-"}
+    _HAS_NAME = True
+    _LOCK = threading.Lock()
+    _MODES = {0: "FM", 1: "AM"}
+    _NOCACHE = "CHIRP_NOCACHE" in os.environ
+    _SPLIT = True
+    _STEPS = (5.0, 6.25, 10.0, 12.5, 15.0, 20.0, 25.0, 30.0, 50.0, 100.0)
+    _TONES = chirp_common.OLD_TONES
+    _LOWER = 0
+    _UPPER = 199
+
+    _CALL_CHANS = ("VHF Call", "UHF Call")
+    _SPECIAL_CHANS = ("L0", "U0",
+                      "L1", "U1",
+                      "L2", "U2",
+                      "L3", "U3",
+                      "L4", "U4",
+                      "L5", "U5",
+                      "L6", "U6",
+                      "L7", "U7",
+                      "L8", "U8",
+                      "L9", "U9",
+                      *_CALL_CHANS)
+
+    # These are used more than once in _COMMON_SETTINGS
+    _BOOL = {'type': 'bool'}
+    _DATA_BAND = {'type': 'list',
+                  'values': ("A", "B", "A:TX/B:RX", "A:RX/B:TX")}
+    _DTMF_MEMORY = {'type': KenwoodD7DTMFMemory}
+    _ON_OFF_DUAL_MAP = {'type': 'map',
+                        'map': (("Off", "0"), ("On", "1,0")),
+                        'set_requires': (('DL', True),),
+                        'get_requires': (('DL', True),)}
+    _POSITION = {'type': KenwoodD7Position}
+    _SSTV_CHARSET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789 !?-/"
+    _SSTV_COLOURS = {'type': 'list',
+                     'values': ("Black", "Blue", "Red", "Magenta", "Green",
+                                "Cyan", "Yellow", "White")}
+    _STATUS_TEXT = {'type': 'string', 'max_len': 32}
+    _SQUELCH = {'type': 'list', 'digits': 2,
+                'values': ("Open", "1", "2", "3", "4", "5")}
+    # Settings that are shared by at least two radios.
+    # If a third radio does not have the setting at all, the presence
+    # here won't matter.  If it has it but with differrent parameters,
+    # it can be overridden in the subclass _SETTINGS dict.
+    _COMMON_SETTINGS = {
+        "AIP": _BOOL,
+        "ARL": {'type': 'map',
+                'map': (('Off', '0000'),) +
+                       tuple([('%i' % x, '%04i' % x)
```

Comment:
I think it's complaining because it thinks this line belongs to the same level as `'map'`, but that would look confusing and weird, so don't do that. Convention would be to put the `tuple(` at the end of the line above and then go in four spaces and I think it'll be happy:
```
'map': (('Off', '0000'),) + tuple(
    [('%i' % x, '%04i' % x)
     for x ...
```

Reply:
Yeah, that takes care of that one, but the other two aren't in a tuple(), they're just straight concatenation.  I'll try going in four spaces instead of visual alignment and see what it does.

Reply:
Still complaining, moving back to what looks right to me for now.